src/Raven.Server/Documents/DatabasesLandlord.cs

```diff
@@ -324,10 +322,22 @@ private Task<DocumentDatabase> CreateDatabaseUnderResourceSemaphore(StringSegmen
         {
             try
             {
-                var task = new Task<DocumentDatabase>(() => ActuallyCreateDatabase(databaseName, config), TaskCreationOptions.RunContinuationsAsynchronously);
-
-                var database = DatabasesCache.GetOrAdd(databaseName, task);
-                if (database == task)
+                var tcs = new TaskCompletionSource<DocumentDatabase>(TaskCreationOptions.RunContinuationsAsynchronously);
```

Comment:
Why aren't we using the much simpler:

` var task = new Task<DocumentDatabase>(() => ActuallyCreateDatabase(databaseName, config), TaskCreationOptions.RunContinuationsAsynchronously);`